relay-server/src/actors/processor.rs

```diff
@@ -2489,6 +2490,62 @@ impl EnvelopeProcessorService {
         Ok(span)
     }
 
+    fn filter_spans(&self, state: &mut ProcessEnvelopeState) {
+        let otel_span_ingestion_enabled =
+            state.project_state.has_feature(Feature::OtelSpanIngestion);
+        state.managed_envelope.retain_items(|item| match item.ty() {
+            ItemType::OtelSpan if !otel_span_ingestion_enabled => {
+                relay_log::warn!("dropping span because feature is disabled");
+                ItemAction::DropSilently
+            }
+            _ => ItemAction::Keep,
+        });
+    }
+
+    #[cfg(feature = "processing")]
+    fn process_spans(&self, state: &mut ProcessEnvelopeState) {
+        let mut items: Vec<Item> = Vec::new();
+        state.managed_envelope.retain_items(|item| match item.ty() {
+            ItemType::OtelSpan => {
+                let span: relay_spans::Span = match serde_json::from_slice(&item.payload()) {
+                    Ok(span) => span,
+                    Err(e) => {
+                        relay_log::error!(error = &e as &dyn Error, "failed to parse OTel span");
+                        return ItemAction::DropSilently;
+                    }
+                };
+                let event_span: Annotated<Span> = Annotated::new(span.into());
+                match self.validate_span(event_span) {
```

Comment:
nit: Could use `let .. else` here.

Reply:
`validate_span` currently relies on certain validations by the `TransactionProcessor`, see https://github.com/getsentry/relay/blob/dda78339e126f9bf6800861efd5ea871aecb80f6/relay-server/src/actors/processor.rs#L2412-L2416. We should replicate this validation somehow (happy to contribute).